lib/src/presentation/view/card_list.dart

```diff
@@ -0,0 +1,341 @@
+    setState(() {
+      actualPage = widget.cardBloc.currentPage.toString();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Scaffold(
+        backgroundColor: Colors.black,
+        appBar: AppBar(
+          title: const Center(
+            child: Text(
+              appBarCardListTitle,
+              style: TextStyle(
+                fontSize: appBarCardListTitleFontSize,
+              ),
+            ),
+          ),
+          backgroundColor: Colors.black,
+        ),
+        endDrawer: Drawer(
+          backgroundColor: Colors.cyanAccent,
+          width: kCardListDrawerWidth,
+          child: ListView(
+            children: [
+              DrawerHeader(
+                decoration: BoxDecoration(
+                  color: Colors.purple,
+                  border: Border.all(
+                    width: kCardListDrawerHeaderWidth,
+                  ),
+                  borderRadius:
+                      BorderRadius.circular(kCardListDrawerHeaderWidth),
+                ),
+                curve: Curves.fastLinearToSlowEaseIn,
+                child: const Text(
+                  kCardListDrawerTitle,
+                  textAlign: TextAlign.center,
+                  style: TextStyle(
+                    fontSize: cardTextInfoFontSize,
+                    fontWeight: FontWeight.bold,
+                  ),
+                ),
+              ),
+              ListTile(
+                title: Title(
+                  color: Colors.black,
+                  child: const Text(
+                    kCardListTotems,
+                    textAlign: TextAlign.center,
+                    style: TextStyle(
+                      fontSize: cardTextInfoFontSize,
+                      fontWeight: FontWeight.bold,
+                    ),
+                  ),
+                ),
+                onTap: () {
+                  _currentUrl = _totemUrl;
+                  widget.cardBloc.sinkCardList(
+                    _currentUrl,
+                  );
+                  updateCurrentPage();
+                  Navigator.pop(
+                    context,
+                  );
+                },
+              ),
+              ListTile(
+                title: Title(
+                  color: Colors.black,
+                  child: const Text(
+                    kCardListMurlocs,
+                    textAlign: TextAlign.center,
+                    style: TextStyle(
+                      fontSize: cardTextInfoFontSize,
+                      fontWeight: FontWeight.bold,
+                    ),
+                  ),
+                ),
+                onTap: () {
+                  _currentUrl = _murlocUrl;
+                  widget.cardBloc.sinkCardList(
+                    _currentUrl,
+                  );
+                  updateCurrentPage();
+                  Navigator.pop(
+                    context,
+                  );
+                },
```

Comment:
You are making the same call on all onTap method, you can abstract this into a function and as parameter you can use the url that you are going to call

Reply:
Done.